amplipi/ctrl.py

```diff
@@ -541,9 +547,35 @@ def _sync_stream_info(self) -> None:
       streams.append(stream)
     self.status.streams = streams
 
+  def _update_serial(self) -> None:
+    if self._serial == None and self.status.info is not None:
+      expanders = []
+      # Preamp Boards
+      eeprom_preamp = EEPROM(BoardType.PREAMP)
+      eeprom_streamer = EEPROM(BoardType.STREAMER_SUPPORT)
+      if eeprom_preamp.present():
+        bi = eeprom_preamp.read_board_info()
+        if bi is not None:
+          self._serial = int(bi.serial)
+          for board in [0x10, 0x18, 0x20, 0x28, 0x30]:
+            if eeprom_preamp.present(board):
+              exp_info = eeprom_preamp.read_board_info(board)
+              if exp_info is not None:
+                expanders.append(exp_info.serial)
+          self._expanders = expanders
+          self.status.info.expanders = expanders
+      elif eeprom_streamer.present():  # Streamers
+        bi = eeprom_streamer.read_board_info()
+        if bi is not None:
+          self._serial = bi.serial
+      else:  # -1 means there is no serial number available from any EEPROM
+        self._serial = -1
+      self.status.info.serial = str(self._serial)
+
   def get_state(self) -> models.Status:
     """ get the system state """
     self._update_sys_info()
+    self._update_serial()
```

Comment:
This looks like it is called every get(). I realize that internally the serial number is cached so that isn't actually what happens.  Why isn't this called in reinit instead? That is where we read a similar property set, the firmware versions.

Reply:
It didn't work in reinit since other things were trying to reach the EEPROM at the same time

Reply:
That's not true, it didn't work in reinit at first because you put it before the preamp's micro was assigned an I2C address, so it couldn't communicate. We didn't realize that until after you had already moved the serial reading to here. There's no reason you couldn't move it back, it just needs to be after the line that calls `rt.Rpi()` to setup the micro/firmware.

Reply:
I tried moving it back to reinit where the firmware is and it stopped working on the regular sized amplipi